src/main/java/org/commcare/xml/SessionDatumParser.java

```diff
@@ -140,6 +134,9 @@ private RemoteQueryDatum parseRemoteQueryDatum()
             } else if ("prompt".equals(tagName)) {
                 String key = parser.getAttributeValue(null, "key");
                 userQueryPrompts.put(key, new QueryPromptParser(parser).parse());
+            } else if ("title".equals(tagName)) {
```

Comment:
@snopoke This commit is similar to how @dominicochoa-dimagi originally wrote the code, and then I asked him to move the title parsing before the loop because I thought that made sense conceptually (since there's at most one title and it'll show up first). Can you say more about why the having the title parsing before the loop doesn't work?

Reply:
It does work as long as the title is present. If it is not present then the parser skips over the next element because the while loop calls `nextTagInBlock` again.

If we want to keep the title handling separate then we could change the while loop into a 'do while' loop.

> since there's at most one title
We could add a check in the loop to make sure the 'title' var is null.

> it'll show up first
It showing up first is purely an artifact of how HQ builds the XML, there's no reason it couldn't come later. I would prefer if the parser did not make assumptions about how the XML is structured where it's not functionally different.

Reply:
ah, I see, the double `nextTagInBlock`. And fair point about the ordering not being a functional part of the suite spec.

(sorry @dominicochoa-dimagi for the misdirection)